File: 1588. Sum of All Odd Length Subarrays/solution.js
/**
 * @param {number[]} arr
 * @return {number}
 */
 var sumOddLengthSubarrays = function(arr) {
  let total = 0;
  let arrLength = arr.length;
  for (let index = 0; index < arr.length; index++){
    let num = arr[index];
    let totalAdded = Math.floor(((index + 1) * (arrLength - index) + 1) / 2);
    total += totalAdded * num;
  }
  return total;
};

File: 1588. Sum of All Odd Length Subarrays/solution.cpp
class Solution {
public:
  int sumOddLengthSubarrays(vector<int>& arr) {
    int total = 0;
    int arrLength = arr.size();
    for (int index = 0; index < arr.size(); index++){
      int num = arr[index];
      int totalAdded = ((index + 1) * (arrLength - index) + 1) / 2;
      total += totalAdded * num;
    }
    return total;
  }
};